src/main/java/com/faforever/client/player/FriendOfflineNotifier.java

```diff
@@ -41,18 +45,24 @@ public void afterPropertiesSet() {
 
   @Subscribe
   public void onUserOnline(UserOfflineEvent event) {
+    NotificationsPrefs notification = preferencesService.getPreferences().getNotification();
     String username = event.getUsername();
     playerService.getPlayerForUsername(username).ifPresent(player -> {
       if (player.getSocialStatus() != SocialStatus.FRIEND) {
         return;
       }
 
-      audioService.playFriendOfflineSound();
-      notificationService.addNotification(
-          new TransientNotification(
-              i18n.get("friend.nowOfflineNotification.title", username), "",
-              IdenticonUtil.createIdenticon(player.getId())
-          ));
+      if (notification.isFriendOfflineSoundEnabled()) {
```

Comment:
wouldn't it be better to do those checks in the audioService methods?

Reply:
I will open up an issue to refactor the whole thing but yes. I just followed the way it was done before and added the missing checks